roles/edpm_network_config/tasks/network_config.yml

```diff
@@ -0,0 +1,149 @@
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: Gather SELinux fact if needed
+  when:
+    - ansible_facts.selinux is undefined
+  ansible.builtin.setup:
+    gather_subset:
+      - "!all"
+      - "!min"
+      - "selinux"
+
+- name: Create fcontext entry for edpmconfig
+  become: true
+  when:
+    - ansible_facts.selinux.status == "enabled"
+  sefcontext:
+    target: "/var/lib/edpm-config(/.*)?"
+    setype: container_file_t
+    selevel: s0
+    state: present
+
+- name: Ensure requirements are satisfied
+  ansible.builtin.include_role:
+    name: edpm_bootstrap
+    tasks_from: packages.yml
+
+- name: Ensure /var/lib/edpm-config directory exists
+  become: true
+  ansible.builtin.file:
+    path: /var/lib/edpm-config
+    state: directory
+    setype: container_file_t
+    selevel: s0
+    mode: 0750
+
+- name: Check for previous run of NetworkConfig
+  become: true
+  ansible.builtin.stat:
+    path: /var/lib/edpm-config/os-net-config.returncode
+  register: os_net_config_returncode_stat
+
+- name: Check result of previous run of NetworkConfig
+  become: true
+  ansible.builtin.slurp:
+    path: /var/lib/edpm-config/os-net-config.returncode
+  when: os_net_config_returncode_stat.stat.exists
+  register: os_net_config_returncode_slurp
+
+- name: Ensure we get cloud-init
+  ansible.builtin.stat:
+    path: /etc/cloud/cloud.cfg.d
+  register: cloud_init_exists
+
+- name: NetworkConfig
+  become: true
+  # The conditions here are when we want to apply the
+  # NetworkConfig. They are:
+  # - If edpm_network_config_update is True
+  # - Or the previous run of NetworkConfig failed.
+  # - Or it has never run
+  # This will match the prior behavior of when a Heat
+  # SoftwareDeployment was used.
+  # It also ensures the script does exist as a sine qua non
+  # condition
+  when:
+    - (edpm_network_config_update) or
+      (os_net_config_returncode_stat.stat.exists and
+      ((os_net_config_returncode_slurp.content | b64decode | int) != 0)) or
+       (not os_net_config_returncode_stat.stat.exists)
+  block:
+    - name: Create /etc/os-net-config directory
+      become: true
+      ansible.builtin.file:
+        path: /etc/os-net-config
+        state: directory
+        recurse: true
+
+    - name: Create os-net-config mappings from lookup data
+      edpm_os_net_config_mappings:
+        net_config_data_lookup:
+          "{{ edpm_network_config_os_net_config_mappings }}"
+      when: not ansible_check_mode|bool
+      register: os_net_config_mappings_result
+
+    - name: Write os-net-config mappings file /etc/os-net-config/mapping.yaml
+      ansible.builtin.copy:
+        content: "{{ os_net_config_mappings_result.mapping | to_nice_yaml }}"
+        dest: /etc/os-net-config/mapping.yaml
+        mode: 0644
```

Comment:
Since this is a refactor, changing the permissions type to string, as per recommendations in ansible documentation [0] may be opportune. 


[0] https://docs.ansible.com/ansible/latest/collections/ansible/builtin/file_module.html#parameter-mode

Reply:
ack!